plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -4766,6 +4848,31 @@ protected Answer execute(UnregisterNicCommand cmd) {
         }
     }
 
+    /**
+     * GetGPUStatsCommand is used to collect the GPU/vGPU details of all the
+     * GPU enabled hosts.
+     * Do Nothing.
+     * @param cmd
+     * @return Answer
+     */
+    protected Answer execute(final GetGPUStatsCommand cmd) {
+        final VmwareContext context = getServiceContext();
+        final VmwareHypervisorHost hyperHost = getHyperHost(context);
+        final HostMO hostMo = new HostMO(context, hyperHost.getMor());
+
+        HashMap<String, HashMap<String, VgpuTypesInfo>> groupDetails = new HashMap<String, HashMap<String, VgpuTypesInfo>>();
+
+        try {
+            groupDetails = hostMo.getGPUGroupDetails();
```

Comment:
What about using the code of line 4873 right after this line?

Reply:
You mean this statement: return new GetGPUStatsAnswer(cmd, groupDetails);?

Reply:
Yes this one. Then, you could do directly inside the Try/catch:
` HashMap<String, HashMap<String, VgpuTypesInfo>> groupDetails = hostMo.getGPUGroupDetails()`

This would avoid the creation of a hashmap that is subsequently just dropped for the garbage collector.